### DISCUSSION TITLE
Efficient processing of large-scale data

### DISCUSSION THREAD

**Discussion by S-Caio** [pydata/xarray#9875] [General]
Hello,

I would like to merge together several tiff files. Each of them corresponds to a province of the Netherlands. They are pretty large files and include 19 different bands. What is the most efficient way of processing/merging these files together? I am currently using rioxaarray.merge to do so but it sometimes throws up memory errors, sometimes it runs. Would love a more dependable method.

Apologies if this is a stupid question; I am new to the geospatial world.

**Comment by scottyhq** [comment]
@S-Caio, as stated your question is too vague to provide input. Please see guidance in this pinned post on creating a discussion topics specific to xarray with examples: https://github.com/pydata/xarray/discussions/5404

You did mention using rioxarray, so the most appropriate place for your question would be in that discussion forum: https://github.com/corteva/rioxarray/discussions
